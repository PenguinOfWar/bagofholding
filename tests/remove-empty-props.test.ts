import removeEmptyProps, {
  isArray,
  isObject
} from '../src/packages/remove-empty-props';

const object = {
  foo: null,
  bar: undefined,
  baz: '',
  fab: false,
  gib: '༼ つ ◕_◕ ༽つ',
  array: [0, 0, 0],
  object: {
    one: 1,
    two: '2',
    three: Number('3'),
    four: null,
    five: undefined,
    six: {},
    seven: []
  }
};

it('recursively removes `null` and `undefined` properties from payload object', () => {
  expect(Object.keys(removeEmptyProps(object)).length).toBe(6);
  expect(Object.keys(removeEmptyProps(object).object).length).toBe(6);
  expect(Object.keys(removeEmptyProps(object)).length).toMatchSnapshot();
});

it('leaves `arrays` untouched', () => {
  expect(isArray(removeEmptyProps(object).array)).toBeTruthy();
  expect(isArray(removeEmptyProps(object).object)).toBeFalsy();
  expect(isObject(removeEmptyProps(object).object)).toBeTruthy();
  expect(isObject(removeEmptyProps(object).array)).toBeTruthy();
  expect(Object.keys(removeEmptyProps(object)).length).toMatchSnapshot();
});
