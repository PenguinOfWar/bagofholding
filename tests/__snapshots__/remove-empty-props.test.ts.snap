// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaves \`arrays\` untouched 1`] = `5`;

exports[`recursively removes \`null\` and \`undefined\` properties from payload object 1`] = `5`;
